natlas-server/app/config_loader.py

```diff
@@ -16,8 +16,7 @@ def load_natlas_services(app, db):
         return
     from app.models import NatlasServices
 
-    current_services = NatlasServices.query.order_by(NatlasServices.id.desc()).first()
-    app.current_services = current_services.as_dict()
+    app.current_services = NatlasServices.get_latest_services()
```

Comment:
Why are we still caching it onto `current_app`, if this is what causes our issues?

Reply:
We're caching it into current_app because /api/getwork still uses it. But it's not caching the actual sqlalchemy object, it's caching a dict representation of it. The problem with AgentScript is that the actual sqlalchemy object is being cached.

I'm working on another PR for getting rid of all of the current_app caching (except ScopeManager which is a bigger discussion because it does actual logic), but I wanted to collect some traces to observe the differences between the current version with caching and the new version without caching.

Reply:
But this does bring up another good point, with this patch alone we never need to cache agentScripts, because `/api/getwork` only uses the `agentScriptsStr` representation of it 🤮. From what I can see, the only place we ever used that cached value was on the page that was causing problems. Everything else that is cached is cached as a dict, not as a list of sqlalchemy objects.